src/main/webapp/app/shared/components/frontend-search/frontend-search.component.ts

```diff
@@ -0,0 +1,41 @@
+import { Component, Input, OnDestroy, OnInit } from '@angular/core';
+import { FormBuilder } from '@angular/forms';
+
+import { Subject } from 'rxjs';
+import { debounce, takeUntil } from 'rxjs/operators';
+import { DEFAULT_DEBOUNCE } from 'app/app.constants';
+
+@Component({
+  selector: 'app-frontend-search',
+  templateUrl: './frontend-search.component.html',
+  styleUrls: ['./frontend-search.component.scss'],
+})
+export class FrontendSearchComponent implements OnInit, OnDestroy {
```

Comment:
Weisst du vom API her stelle ich mir das so vor:
Hier ist ein FormControl drin (das eigentliche filter field) dieses hört auf seine value changes und pusht die direkt auf ein (filter) Output. Von aussen ist das dann sehr einfach zu verbauen, und man erhält direkt den Wert.

@Output
filter = new EventEmitter<string()>;

this.formControl.valueChanges
.pipe ..
.subscribe(this.filter); 

Von der usage ists dann <filter-field (filter)=onFilter($event)></filter-field>

Reply:
e.g.
```html
<app-frontend-search class="float-left"
                                 (valueChange)="this.filterData($event)"></app-frontend-search>
```
and
```ts
filterData(searchString: string): void {
  this.responsibilities = this.loadedResponsibilities!.filter(
    (responsibility: IResponsibility) => responsibility.name?.indexOf(searchString) !== -1
  );
}
```